IPyth.sol

```diff
@@ -74,6 +74,26 @@ interface IPyth {
     /// @param updateData Array of price update data.
     function updatePriceFeeds(bytes[] memory updateData) external payable;
 
+    /// @notice Wrapper around updatePriceFeeds that rejects fast if a price update is not needed based on the
```

Comment:
I wonder if instead of requiring the user to pass in an array of observed times, we could just parse the vm, instead of verifying it, so we could perform the time check before the VAA verification?

Reply:
The reason I'm doing it like this is reduce gas usage as much as possible, I think even parsing the VAA has a lot of costs (as it creates a new struct).

Also, it's not only to check this vaa existed or not. The original purpose of this is to allow multiple price pushers without a lot of gas overhead.

They don't necessarily care about that specific VAA being there. They know evm price X and observed price Y, they can do any of these two:
1) update the price only if the contract is X (they submit `X.publishTime`)
2) update the price only if the contract has price older than Y (they submit `Y.publishTime - 1`). 

VAA check is similar to second approach. However, for the price pushers the first one is enough, as when they see the update to price X (which is not Y) they might not want to push price Y at all. So it's more efficient and is different.

Also its quite possible that second check do not be effective at all when we increase our update frequency to 1 second. I mean multiple price pushers actually will change the price successfully whereas only 1 is enough for their purpose.

Reply:
Update:
I changed the code and the docs so the condition is stricly `<` so it implies the first approach as it is easier to understand. However I will use the second approach in the price pusher for optimization (and it will submit `X.publishTime + 1`)

cc: @jayantk